### PR TITLE
feature/PIMOB-2262_Artifact_generation_and_download_via_GHA

### PR DESCRIPTION
**Pull request by fabio-insolia-cko**
## Issue

PIMOB-2262

## Proposed changes

Simplification of the general flows; loads of jobs (unit test, assemble and lint) were always enabled for all the flows, so I have removed the redundant flag.
While doing the assemble task, I have introduced a new boolean flag to:
- create both debug and release builds
- provide the output of those available on github action

Screenshot 2023-11-27 at 17 19 07

## Test Steps

1. Do a change on checkout module
2. verify that all the job will trigger
3. Verify that only app and example-app flow will have artifacts to download

## Checklist

* [x] Reviewers assigned
* [x] I have performed a self-review of my code and manual testing
* [x] Lint and unit tests pass locally with my changes

### CODE REVIEW

File: .github/workflows/cleanup-artifacts.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    - cron: '0 1 * * 1'
+
+jobs:
+  remove-old-artifacts:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+
+    steps:
+      - name: Clean all artifacts
+        uses: c-hive/gha-remove-artifacts@v1
```
Comment (jheng-hao-lin-cko) [inline .github/workflows/cleanup-artifacts.yml:17]: Hmmm this action doesn't seem to be created by a verified user and offiical account. We would need to request a secruity check via freshservice before using it.

Comment (fabio-insolia-cko) [reply .github/workflows/cleanup-artifacts.yml]: I will remove the flow for now and then, once we verify better, reintroduce it

**Review by chintan-soni-cko — COMMENTED** [review]
@fabio-insolia-cko Quick question: I have downloaded the artifacts. It only contains the apks from the example_app_frames and not from the app module. Do I miss something here?

**Comment by fabio-insolia-cko** [comment]
> @fabio-insolia-cko Quick question: I have downloaded the artifacts. It only contains the apks from the example_app_frames and not from the app module. Do I miss something here?

https://github.com/checkout/frames-android/actions/runs/7008300534
https://github.com/checkout/frames-android/actions/runs/7008300564
both flows are from the same pull but one build the example app and the other build the other app
Happy to provide further explanation offline

**Review by chintan-soni-cko — APPROVED** [review]

**Pull request merged**